src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/RemoveAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,166 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System;
+
+namespace Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlDatabaseInstanceFailoverGroup", DefaultParameterSetName = RemoveIFGDefaultSet,
+        SupportsShouldProcess = true), OutputType(typeof(AzureSqlInstanceFailoverGroupModel))]
+    public class RemoveAzureSqlInstanceFailoverGroup : AzureSqlInstanceFailoverGroupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for the default remove.
+        /// </summary>
+        private const string RemoveIFGDefaultSet = "RemoveIFGDefault";
+
+        /// <summary>
+        /// Parameter set name for remove with an Input Object.
+        /// </summary>
+        protected const string RemoveIFGByInputObjectParameterSet =
+            "Remove a Instance Failover Group from AzureSqlInstanceFailoverGroupModel instance definition";
+        
+        /// <summary>
+        /// Parameter set name for remove with a resource ID.
+        /// </summary>
+        private const string RemoveIFGByResourceIdSet = "Remove a Instance Failover Group from Resource Id";
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveIFGDefaultSet, 
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the local region to use.
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveIFGDefaultSet, 
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the Primary Region of the Instance Failover Group.")]
+        [LocationCompleter("Microsoft.Sql/locations/instanceFailoverGroups")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the InstanceFailoverGroup to remove.
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveIFGDefaultSet, 
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "The name of the Azure SQL Database Instance Failover Group to remove.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+		/// Instance Failover Group object to remove
+		/// </summary>
+		[Parameter(ParameterSetName = RemoveIFGByInputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The Instance Failover Group object to remove")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlInstanceFailoverGroupModel InputObject { get; set; }
```

Comment:
I was told to add the InputObject and ResourceId in the PR I had originally started in the preview branch (https://github.com/Azure/azure-powershell/pull/5932). Since I assume that eventually I need to check into that branch, should I keep those changes?